p2p/transport/websocket/addrs.go

```diff
@@ -10,52 +10,28 @@ import (
 	manet "github.com/multiformats/go-multiaddr/net"
 )
 
-// Addr is an implementation of net.Addr for WebSocket.
-type Addr struct {
+// addrWrapper is an implementation of net.Addr for WebSocket.
+type addrWrapper struct {
 	*url.URL
 }
 
-var _ net.Addr = (*Addr)(nil)
+var _ net.Addr = addrWrapper{}
```

Comment:
Why is this one needed

Reply:
This is a pretty standard compile time interface assertion.